pkg/apis/flyteworkflow/v1alpha1/node_status.go

```diff
@@ -517,62 +518,53 @@ func (in NodeStatus) GetTaskNodeStatus() ExecutableTaskNodeStatus {
 	return in.TaskNodeStatus
 }
 
-func (in *NodeStatus) GetNodeExecutionStatus(ctx context.Context, id NodeID) ExecutableNodeStatus {
-	n, ok := in.SubNodeStatus[id]
-	if ok {
-		n.SetParentTaskID(in.GetParentTaskID())
-		n.DataReferenceConstructor = in.DataReferenceConstructor
-		if len(n.GetDataDir()) == 0 {
-			dataDir, err := in.DataReferenceConstructor.ConstructReference(ctx, in.GetDataDir(), id)
-			if err != nil {
-				logger.Errorf(ctx, "Failed to construct data dir for node [%v]", id)
-				return n
-			}
-
-			n.SetDataDir(dataDir)
+func (in *NodeStatus) setEphemeralNodeExecutionStatusAttributes(ctx context.Context, id NodeID, n *NodeStatus) error {
+	n.SetParentTaskID(in.GetParentTaskID())
+	if len(n.GetDataDir()) == 0 {
+		dataDir, err := in.DataReferenceConstructor.ConstructReference(ctx, in.GetOutputDir(), id)
+		if err != nil {
+			return fmt.Errorf("failed to construct data dir for node [%v]. Error: %w", id, err)
 		}
 
-		if len(n.GetOutputDir()) == 0 {
-			outputDir, err := in.DataReferenceConstructor.ConstructReference(ctx, n.GetDataDir(), strconv.FormatUint(uint64(in.Attempts), 10))
-			if err != nil {
-				logger.Errorf(ctx, "Failed to construct output dir for node [%v]", id)
-				return n
-			}
+		n.SetDataDir(dataDir)
+	}
 
-			n.SetOutputDir(outputDir)
+	if len(n.GetOutputDir()) == 0 {
+		outputDir, err := in.DataReferenceConstructor.ConstructReference(ctx, n.GetDataDir(), strconv.FormatUint(uint64(n.Attempts), 10))
```

Comment:
why is this using n.GetDataDir to set the outputDir and above in line 524 the OutputDir to set the dataDir. This seems backwards

Reply:
since the datadir is already set, output dir is just a derivative from that